src/lib/handleFileUpload.ts

```diff
@@ -18,8 +18,11 @@ export const handleFileUpload = async (
 
   state.setCollection(collection);
 
+  // NOTE: we can't use entry.name === encodeURI(file.name) here,
+  // because server and encodeURI() might use different case for
+  // %-encoded values, like "%D0" vs "%d0"
```

Comment:
I think this won't work because `toUpperCase()`/`toLowerCase()` will also change case of other letters  in filename. For example, if server says that it has a file called `A%2dFile.txt`, and we try to upload a file called `A%2DFile.txt` (different case of letter "D"), then `toUpperCase()` and `toLowerCase()` will turn it to `A%2DFILE.TXT` and `a%2dfile.txt`, respectively, which won't match filename on the server.

We could convert both filenames to upper case, but then we would assume that backend server case-insensitive, and uploading a file called `FILE.TXT` overwrites a file called `file.txt`. Which is actually a case only on Windows.

Reply:
Absolutely. Another alternative could be decoding and re-encoding so that the case (should?) be consistent as it's using the browser to perform the encode both sides. For `.toUpperCase()` we could `.replace(/%[0-9a-f]{2}/g, (s) => s.toUpperCase())` although I wonder if we'll encounter problems with values like `%252d`/`%252D` etc.